src/convex/ConvexAvatarMultiToken.sol

```diff
@@ -0,0 +1,832 @@
+    function setMinOutBpsVal(address _tokenA, address _tokenB, uint16 _minOutBpsVal) external onlyOwnerOrManager {
+        if (_minOutBpsVal > MAX_BPS) {
+            revert InvalidBps(_minOutBpsVal);
+        }
+
+        uint256 minOutBpsMin = minOutBps[_tokenA][_tokenB].min;
+        if (_minOutBpsVal < minOutBpsMin) {
+            revert LessThanBpsMin(_minOutBpsVal, minOutBpsMin);
+        }
+
+        uint256 oldMinOutBpsVal = minOutBps[_tokenA][_tokenB].val;
+        minOutBps[_tokenA][_tokenB].val = _minOutBpsVal;
+
+        emit MinOutBpsValUpdated(_tokenA, _tokenB, oldMinOutBpsVal, _minOutBpsVal);
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    // PUBLIC: Owner
+    ////////////////////////////////////////////////////////////////////////////
+
+    /// @notice Takes a given pid to create a private vault for the avatar as owner. Can only be called by owner.
+    /// @dev Keeps in storage the `pid` and `vaultAddr` created during execution
+    /// @param _pid Pid target to create a private vault to stake into
+    function createPrivateVault(uint256 _pid) external onlyOwner {
+        _createPrivateVault(_pid);
+    }
+
+    /// @notice Takes a given amount of asset from the owner and stakes them on private vault. Can only be called by owner.
+    /// @param _pid Pid target to stake into its appropiate vault
+    /// @param _amountAsset Amount of asset to be lock.
+    /// @param _additionalFunds Allows to deposit further funds in the same lock within the private vault when `True`
+    function depositInPrivateVault(uint256 _pid, uint256 _amountAsset, bool _additionalFunds)
+        external
+        onlyOwner
+        returns (bytes32 kekId)
+    {
+        if (_amountAsset == 0) {
+            revert NothingToDeposit();
+        }
+
+        address vaultAddr = privateVaults[_pid];
+        if (vaultAddr == address(0)) {
+            revert NoPrivateVaultForPid(_pid);
+        }
+
+        IStakingProxy proxy = IStakingProxy(vaultAddr);
+        address stakingToken = proxy.stakingToken();
+        IFraxUnifiedFarm farm = IFraxUnifiedFarm(proxy.stakingAddress());
+
+        IERC20MetadataUpgradeable(stakingToken).safeTransferFrom(msg.sender, address(this), _amountAsset);
+
+        if (_additionalFunds) {
+            kekId = kekIds[vaultAddr];
+            if (kekId == bytes32(0)) {
+                revert NoExistingLockInPrivateVault(vaultAddr);
+            }
+
+            proxy.lockAdditional(kekId, _amountAsset);
+        } else {
+            // NOTE: we always try to lock for the min duration allowed
+            uint256 minLockTime = farm.lock_time_min();
+            if (minLockTime > MAX_LOCKING_TIME) {
+                revert MinLockTimeMisconfigured(address(farm), minLockTime);
+            }
+            kekId = proxy.stakeLocked(_amountAsset, farm.lock_time_min());
+            /// @dev detailed required to enable withdrawls later
+            kekIds[vaultAddr] = kekId;
```

Comment:
Same issue with CEI

Reply:
This may be the one time you must use a nonReentrant modifier (if necessary obv)

```
            proxy.lockAdditional(kekId, _amountAsset);
        } else {
            // NOTE: we always try to lock for the min duration allowed
            uint256 minLockTime = farm.lock_time_min();
            if (minLockTime > MAX_LOCKING_TIME) {
                revert MinLockTimeMisconfigured(address(farm), minLockTime);
            }
            kekId = proxy.stakeLocked(_amountAsset, farm.lock_time_min());
            /// @dev detailed required to enable withdrawls later
            kekIds[vaultAddr] = kekId;
        }
```

Reply:
considerations while double checking reentrancy concern:

- method itself is permissioned to the owner, in this case the avatar
- contract does have already in [stakeLocked](https://github.com/convex-eth/frax-cvx-platform/blob/main/contracts/contracts/StakingProxyConvex.sol#L92) a `nonReentrant` modifier
- does not present fallback methods
- approvals from avatar's owner will be limited to the `_amountAsset` per atomic tx ideally

re CEI as discussed in dms this particular case, it is not possible to be cei compliant due to the dependancy of the `kekId` from the external call and after we write into storage

will drop a commit with minor comments to avoid other falling into the same concern in the same L

Reply:
comments on DD in c908449d96a23e6a9808edceff7116d773790a2d